spec/services/remove_cps_service_spec.rb

```diff
@@ -0,0 +1,97 @@
+RSpec.describe RemoveCpsService do
+  let(:tags) { { a: { any: %w[b] } } }
+  let(:content_purpose_supergroup) { 'news_and_communications' }
+  let(:updated_tags) { tags.merge(content_purpose_supergroup: { any: [content_purpose_supergroup] }) }
+
+  describe '#deactivate_subscriber_lists' do
+    it 'does not create a new SubscriberList because it does not contain a CPS' do
+      subscriber_list = FactoryBot.create(:subscriber_list)
+      expect { RemoveCpsService.new.deactivate_subscriber_lists }.to_not(change { subscriber_list.reload.attributes })
+    end
+    it 'removes the CPS from the subscriber list' do
+      subscriber_list = FactoryBot.create(:subscriber_list, content_purpose_supergroup: content_purpose_supergroup)
+      RemoveCpsService.new.deactivate_subscriber_lists
+      subscriber_list.reload
+      expect(subscriber_list.content_purpose_supergroup).to be nil
+    end
+    it 'adds the content purpose supergroup to the tags' do
+      subscriber_list = FactoryBot.create(:subscriber_list, tags: tags, content_purpose_supergroup: content_purpose_supergroup)
+      RemoveCpsService.new.deactivate_subscriber_lists
+      subscriber_list.reload
+      expect(subscriber_list.tags).to include(content_purpose_supergroup: { any: [content_purpose_supergroup] })
+      expect(subscriber_list.tags).to include(tags)
+    end
+    context 'A duplicate subscriber list exists' do
+      before :each do
+        @subscriber_list = FactoryBot.create(:subscriber_list_with_subscribers,
+                                             tags: tags,
+                                             content_purpose_supergroup: content_purpose_supergroup,
+                                             subscriber_count: 2)
+        @duplicate_subscriber_list = FactoryBot.create(:subscriber_list, tags: updated_tags, content_purpose_supergroup: nil)
+      end
+      it 'adds the subscriptions to the duplicate' do
+        expect { RemoveCpsService.new.deactivate_subscriber_lists }.to change {
+          @duplicate_subscriber_list.reload.subscribers.count
```

Comment:
Might also be worth checking that the subscribers have been moved off @subscriber_list, or do we not mind about this?